engine/runtime/src/main/java/org/enso/interpreter/runtime/TruffleCompilerContext.java

```diff
@@ -374,12 +375,19 @@ public QualifiedName getName() {
     @Override
     public BindingsMap getBindingsMap() {
       if (module.getIr() != null) {
-        var meta = module.getIr().passData();
-        var pass = meta.get(BindingAnalysis$.MODULE$);
-        return (BindingsMap) pass.get();
-      } else {
-        return bindings;
+        try {
+          var meta = module.getIr().passData();
+          var pass = meta.get(BindingAnalysis$.MODULE$);
+          emitIOException();
+          return (BindingsMap) pass.get();
+        } catch (IOException ex) {
```

Comment:
Lazy deserialization of `BindingsMap` may fail: https://github.com/enso-org/enso/actions/runs/6974085571/job/18979188994?pr=8359#step:10:1073 - don't propagate the `IOException` further.